index.rst

```diff
@@ -143,104 +164,207 @@ Core Components
 Sensor Components
 -----------------
 
+Sensors have been split into categories. If a sensor fits into more than one category, it will be listed multiple times.
+
+Core / Internal
+***************
 .. imgtable::
 
     Sensor Core, components/sensor/index, folder-open.svg
-    ADC, components/sensor/adc, flash.svg
-    ADE7953, components/sensor/ade7953, ade7953.svg
-    ADS1115, components/sensor/ads1115, ads1115.jpg
-    AHT10, components/sensor/aht10, aht10.jpg
-    AM2320, components/sensor/am2320, am2320.jpg
-    APDS9960, components/sensor/apds9960, apds9960.jpg
-    ATM90E32, components/sensor/atm90e32, atm90e32.jpg
-    AS3935, components/sensor/as3935, as3935.jpg
-    BH1750, components/sensor/bh1750, bh1750.jpg
-    Binary Sensor Map, components/sensor/binary_sensor_map, binary_sensor_map.jpg
-    BLE RSSI, components/sensor/ble_rssi, bluetooth.svg
-    BLE Client Sensor, components/sensor/ble_client, bluetooth.svg
-    BME280, components/sensor/bme280, bme280.jpg
-    BME680, components/sensor/bme680, bme680.jpg
-    BME680 via BSEC, components/sensor/bme680_bsec, bme680.jpg
-    BMP085, components/sensor/bmp085, bmp180.jpg
-    BMP280, components/sensor/bmp280, bmp280.jpg
-    b-parasite, components/sensor/b_parasite, b_parasite.jpg
-    CCS811, components/sensor/ccs811, ccs811.jpg
-    CSE7766, components/sensor/cse7766, cse7766.svg
-    CS5460A, components/sensor/cs5460a, cs5460a.png
-    CT Clamp, components/sensor/ct_clamp, ct_clamp.jpg
-    Dallas, components/sensor/dallas, dallas.jpg
-    DHT, components/sensor/dht, dht.jpg
-    DHT12, components/sensor/dht12, dht12.jpg
-    Duty Cycle, components/sensor/duty_cycle, percent.svg
-    ESP32 Hall Sensor, components/sensor/esp32_hall, magnet.svg
-    EZO sensor circuits, components/sensor/ezo, ezo-ph-circuit.png
-    Havells Inverter, components/sensor/havells_inverter, havellsgti5000d_s.jpg
-    HDC1080, components/sensor/hdc1080, hdc1080.jpg
-    HLW8012, components/sensor/hlw8012, hlw8012.svg
-    HMC5883L, components/sensor/hmc5883l, hmc5883l.jpg
-    QMC5883L, components/sensor/qmc5883l, qmc5883l.jpg
+    Custom Sensor, components/sensor/custom, language-cpp.svg
```

Comment:
I believe this section should be ordered this way:
- Core
- all internal sensors, like Home Assistant, resistance, etc. (alphabetically may be)
- Template
- Custom

This logic could be expanded to other sections

Reply:
agree, but moved ADC and Resistance to analogue

Reply:
> I agree that the sensors section has become too big and it'd be nice to split it up, but the current implementation feels very cluttered to me.
> 
> Part of this are the headings in the TOC, which are long, inconsistent and have lots of parentheses and slashes. I think some of the following substitutions will help there (though others might have better ideas):
> 
> * Core / Internal -> Core (all sensors on the site are internal, right?)
> * Air Quality / Gas -> Air
> * Ambient Light / Colour -> Light (also I think we use American English on the site)
> * Analogue To Digital -> Analogue (resistance should also be here and not under Core)
> * Load Cell (Force / Weight) -> Weight
> * Magnetic / Hall Effect -> Magnetic
> * Motion, Gesture -> Motion
> * Current / Power / Energy -> Electricity
> * Solar (PV) Inverters -> it's just one, put it under misc
> * Temperature / Humidity / Pressure -> not sure how to name this (environment is probably too generic), but it's the worst one imo. Weather sort of makes sense except most of these are used indoor 😆
> 
> The other part of this are the sensor functions below their names, which have lots of abbreviations and ampersands. I'm not sure how to best solve that though (I'm not a designer either xD). Usually I'd prefer icons for things like that, but I don't think that works here.
> 
> Some suggestions though (not everyone might agree with these):
> 
> * Don't overuse capitals for every word.
> * Either put a description for all sensors on a row, or none, but don't mix it
> * ESP internal -> Built-in or ESP built-in
> * STS3X -> this should be under Temp/Humidity/Pressure
> * X-Ch -> X-channel
> * A/D -> ADC
> * HX711: Amp -> amplifier
> * EZO: Power Of Hydrogen can be left out I think, I don't know anyone who calls it that as opposed to pH
> * Thermocouple -> I don't think it's relevant what the communication type is, none of the other sensors have it listed

thanks @oxan I like the simplified headings - done. Also made the other changes you suggested.
Regarding sensor name tags, I used "&" because the imgtable:: RST directive uses CSV, so it breaks using commas. I could use "-" or "/" or just spaces to seprate multiple sensor functions. I will commit all the other changes now, let me know preferred approach. I really like the icon idea, maybe a separate pull request by a graphic designer!